iocp/data/patterns.ini

```diff
@@ -7,19 +7,19 @@ pattern:	\b(([a-z0-9\-]{2,}\[?\.\]?)+(abogado|ac|academy|accountants|active|acto
 defang:		True
 
 [IP]
-pattern:	\b(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\b
+pattern:	\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$\b
 
 [Email]
-pattern:	\b([a-z][_a-z0-9-.]+@[a-z0-9-]+\.[a-z]+)\b
+pattern:	\b([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$\b
 
 [MD5]
-pattern:	\b([a-f0-9]{32}|[A-F0-9]{32})\b
+pattern:	\b[a-fA-F0-9]{32}(:.+)?$\b
 
 [SHA1]
-pattern:	\b([a-f0-9]{40}|[A-F0-9]{40})\b
+pattern:	\b[a-fA-F0-9]{40}(:.+)?$\b
 
 [SHA256]
-pattern:	\b([a-f0-9]{64}|[A-F0-9]{64})\b
+pattern:	\b[a-fA-F0-9]{64}(:.+)?$\b
```

Comment:
ok but need to remove the $ at the end in order to match the SHA256 in any part of the document and not only at the end of the line (example: 567efd7abb99428737d22bf3f8cee9d23a540b4e1565938b557eec54078b8a31|putty.exe)